model/src/main/java/com/flipkart/flux/domain/AuditRecord.java

```diff
@@ -109,28 +109,30 @@ public Timestamp getCreatedAt() {
     }
 
     @Override
-    public int hashCode() {
-        final int prime = 31;
-        int result = 1;
-        result = prime * result + ((id == null) ? 0 : id.hashCode());
-        return result;
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof AuditRecord)) return false;
+
+        AuditRecord that = (AuditRecord) o;
+
+        if (retryAttempt != that.retryAttempt) return false;
```

Comment:
why are we using = & != ?
This just checks if both references are same.
We must use .equals() of the objects. Also, check if nullable objects are null before performing .equals() else it'll bomb. Use intellij to generate these for you